internal/campaigns/bind_workspace_test.go

```diff
@@ -329,10 +330,15 @@ func isDir(t *testing.T, path string) bool {
 	return st.IsDir()
 }
 
-func readWorkspaceFiles(workspace string) (map[string]string, error) {
+func readWorkspaceFiles(workspace Workspace) (map[string]string, error) {
+	w, ok := workspace.(*dockerBindWorkspace)
+	if !ok {
+		return nil, errors.Errorf("unknown workspace of type %T: %+v", workspace, workspace)
+	}
+
 	files := map[string]string{}
 
-	err := filepath.Walk(workspace, func(path string, info os.FileInfo, err error) error {
+	err := filepath.Walk(w.dir, func(path string, info os.FileInfo, err error) error {
```

Comment:
Do you even need that assertion? I think with the interface in place you could simply call `w.WorkDir()` and walk that, right?

(I'd also understand if you want to assert the type here, so feel free to ignore this comment)

Reply:
No, that's betraying `WorkDir()` being a relatively late addition to the interface. I'll change that.